src/middlewared/middlewared/plugins/ftp.py

```diff
@@ -197,8 +197,13 @@ async def do_update(self, data):
         if new["onlyanonymous"]:
             if not new["anonpath"]:
                 verrors.add("ftp_update.anonpath", "This field is required for anonymous login")
-            else:
-                await check_path_resides_within_volume(verrors, self.middleware, "ftp_update.anonpath", new["anonpath"])
+        else:
+            # Anonymous is disabled, clear the anonpath
+            if new["anonpath"] is not None:
+                new["anonpath"] = None
```

Comment:
So if anonymous ftp access is turned off and then turned back on again, then the administrator **must** to re-select a path?  Personally that's not the behavior that I'd prefer, but I suppose an argument could be made either way.